src/services/pegout-data.processor.ts

```diff
@@ -267,14 +267,25 @@ export class PegoutDataProcessor implements FilteredBridgeTransactionProcessor {
   }
 
   private async processReleaseRequestReceivedStatus(extendedBridgeTx: ExtendedBridgeTx): Promise<void> {
-
     const events: BridgeEvent[] = extendedBridgeTx.events;
     const releaseRequestReceivedEvent = events.find(event => event.name === BRIDGE_EVENTS.RELEASE_REQUEST_RECEIVED);
 
     if(!releaseRequestReceivedEvent) {
       return;
     }
 
+    const status = await this.fillRequestReceivedStatus(extendedBridgeTx, releaseRequestReceivedEvent);
+
+    try {
+      await this.pegoutStatusDataService.set(status);
+      this.logger.trace(`[processReleaseRequestReceivedStatus] ${extendedBridgeTx.txHash} registered`);
+    } catch(e) {
+      this.logger.warn('[processReleaseRequestReceivedStatus] There was a problem with the storage', e);
+    }
+
+  }
+
+  public async fillRequestReceivedStatus(extendedBridgeTx: ExtendedBridgeTx, releaseRequestReceivedEvent: BridgeEvent):Promise<PegoutStatusDbDataModel> {
```

Comment:
The extendedBridgeTx already includes the releaseRequestReceivedEvent, so, let's only use the extendedBridgeTx. Also, I suggest changing the function signature to:

public static async getRequestReceivedStatus(extendedBridgeTx: ExtendedBridgeTx):Promise<PegoutStatusDbDataModel> ...

Maybe put getRequestReceivedStatus in another file like Jose suggested?